packages/design-toolkit/src/components/coordinate-field/index.tsx

```diff
@@ -0,0 +1,463 @@
+          >
+            {labelProp}
+          </Label>
+        )}
+
+        <div
+          className={control({ className: classNames?.control })}
+          style={{ width: maxControlWidth }}
+        >
+          <div
+            className={input({ className: classNames?.input })}
+            onPasteCapture={paste.handleInputPaste}
+            data-input-container
+          >
+            {state.segmentConfigs.map((config, configIndex) => {
+              if (config.type === 'literal') {
+                return (
+                  <span
+                    key={`${format}-literal-${configIndex}-${config.value}`}
+                    className='fg-primary-muted'
+                  >
+                    {config.value}
+                  </span>
+                );
+              }
+
+              const editableIndex = state.segmentConfigs
+                .slice(0, configIndex)
+                .filter((c) => c.type !== 'literal').length;
+
+              return (
+                <CoordinateSegment
+                  key={`${format}-segment-${editableIndex}`}
+                  value={state.segmentValues[editableIndex] || ''}
+                  onChange={(newValue) =>
+                    state.handleSegmentChange(editableIndex, newValue)
+                  }
+                  onFocus={() => focus.setFocusedSegmentIndex(editableIndex)}
+                  onBlur={() => {
+                    focus.setFocusedSegmentIndex(-1);
+                    state.flushPendingValidation();
+                  }}
+                  onKeyDown={(e) =>
+                    focus.handleSegmentKeyDown(editableIndex, e)
+                  }
+                  onAutoAdvance={() => focus.focusNextSegment(editableIndex)}
+                  onAutoRetreat={() =>
+                    focus.focusPreviousSegment(editableIndex)
+                  }
+                  placeholder={config.placeholder}
+                  maxLength={config.maxLength}
+                  pad={config.pad}
+                  className={segment({ className: classNames?.segment })}
+                  isDisabled={isDisabled}
+                  allowedChars={config.allowedChars}
+                  segmentRef={focus.segmentRefs[editableIndex]}
+                  segmentIndex={editableIndex}
+                  totalSegments={state.editableSegmentConfigs.length}
+                  ariaLabel={getSegmentLabel(format, editableIndex)}
+                />
+              );
+            })}
+          </div>
+
+          {showFormatButton && (
+            <Popover onOpenChange={handlePopoverOpenChange}>
+              <PopoverTrigger>
+                <Button
+                  variant='icon'
+                  size={size}
+                  color='mono-bold'
+                  className={classNames?.formatButton}
+                  aria-label='View coordinate in all formats'
+                  isDisabled={!copy.isFormatButtonEnabled}
+                  data-format-button
```

Comment:
interesting: I'm more used to seeing `data-testid='format-button'` instead of custom data properties - at least in application code. I don't think it's an antipattern at all, just different than what I'm used to seeing. A search through the design-toolkit reveals that we're using data-size and other custom data- fields, so may just be my own unfamiliarity of the pattern in dtk.